templates/ecs.yaml

```diff
@@ -0,0 +1,317 @@
+  DesiredCapacity:
+    Type: Number
+    Default: '1'
+    Description: Number of instances to launch in your ECS cluster.
+  MaxSize:
+    Type: Number
+    Default: '1'
+    Description: Maximum number of instances that can be launched in your ECS cluster.
+  InstanceType:
+    Description: EC2 instance type
+    Type: String
+    Default: t3a.micro
+    AllowedValues: [t3.micro, t3.small, t3.medium, t3.large, t3a.micro, t3a.small, t3a.medium, t3a.large]
+    ConstraintDescription: Please choose a valid instance type.
+  DockerImageTag:
+    Type: String
+    Description: The Docker image tae to run
+Mappings:
+  AWSRegionToAMI:
+    us-east-1:
+      AMIID: ami-0669eafef622afea1
+    us-west-2:
+      AMIID: ami-05d33de432484ef34
+Resources:
+  # iAtlas API task execution IAM role
+  # This is necessary so it can read the Secrets Manager entry that contains the external
+  # registry credentials
+  IAtlasIAMExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ecs-tasks.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      Policies:
+        - PolicyName: AccessRegCredSecret
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - ssm:GetParameters
+                  - secretsmanager:GetSecretValue
+                  - kms:Decrypt
+                Resource:
+                  - !Sub 'arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/*'
+                  - !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:*'
+                  - !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/*'
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogStream
+                Resource: '*'
+  ECSCluster:
+    Type: AWS::ECS::Cluster
+
+  # Security group for ECS nodes
+  # Should allow ephemeral ports from the ALB SG
+  # Should allow SSH from the Bastion hosts security group
+  EcsSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: ECS Node Security Group
+      VpcId: !Ref 'VpcId'
+  EcsSecurityGroupALBports:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref 'EcsSecurityGroup'
+      IpProtocol: tcp
+      FromPort: 31000
+      ToPort: 61000
+      SourceSecurityGroupId: !Ref 'EcsAlbSecurityGroup'
+  EcsSecurityGroupSSHfromBastion:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref 'EcsSecurityGroup'
+      IpProtocol: tcp
+      FromPort: 22
+      ToPort: 22
+      SourceSecurityGroupId: !Ref 'BastionHostSgId'
+
+  # Security group for the ALB that goes in front
+  # Should only allow port 80 from the world for now
```

Comment:
is your plan to leave adding port 443 at a later time?

Reply:
I thought we had discussed this, but we have no intent on adding port 443, because the API is essentially read-only and is only presenting data that's already in the public domain. We don't yet know how the DNS for this is going to be handled, so there's not really anyway to automate certificate management either.

Reply:
do you have an opinion about this @ahayden ?  i'm assuming we want to support `https` which would mean port 443, correct?

Reply:
Yes, this should probably be 443, and the ECS interface needs to force SSL/TLS. A self-signed cert is fine. The reason for this a security/compliance requirement of this AWS account